### ISSUE TITLE
Difference between ResNet and VGG train.prototxt?

### ISSUE THREAD

**Issue by LamForest** [cvjena/cnn-models#17]
In VGG, ReLU follows BatchNorm Layer immediately but in ResNet, a Scale Layer is inserted between BatchNorm and ReLU layer. Why is the inconsistency? What is the difference between these two usage?

**Comment by LamForest** [comment]
According to Caffe Documents:
http://caffe.berkeleyvision.org/doxygen/classcaffe_1_1BatchNormLayer.html#details
> Normalizes the input to have 0-mean and/or unit (1) variance across the batch.
This layer computes Batch Normalization as described in [1]. For each channel in the data (i.e. axis 1), it subtracts the mean and divides by the variance, where both statistics are computed across both spatial dimensions and across the different examples in the batch.
By default, during training time, the network is computing global mean/variance statistics via a running average, which is then used at test time to allow deterministic outputs for each input. You can manually toggle whether the network is accumulating or using the statistics via the use_global_stats option. For reference, these statistics are kept in the layer's three blobs: (0) mean, (1) variance, and (2) moving average factor.
Note that the original paper also included a per-channel learned bias and scaling factor. **To implement this in Caffe, define a ScaleLayer configured with bias_term: true after each BatchNormLayer to handle both the bias and scaling factor.**

So I think a BatchNorm Layer is always used with a scale Layer.

**Comment by MarcelSimon** [comment]
Hi Tianlin Gao,
please see the similar issue https://github.com/cvjena/cnn-models/issues/11
